BSP/STM32F413/Src/BSP_SPI.c

```diff
@@ -126,3 +133,88 @@ void BSP_SPI_SetStateCS(uint8_t state) {
         GPIO_ResetBits(GPIOB, GPIO_Pin_6);
     }
 }
+
+/***************THE FOLLWING CODE IS FOR THE RTOS VERSION OF THE BPS*********/
+#ifdef RTOS
+
+void SPI1_IRQHandler(void){
+	OS_ERR err;
+	// Save the CPU registers
+	CPU_SR_ALLOC();
+
+	// Protect a critical section
+	CPU_CRITICAL_ENTER();
+
+	// make the kernel aware that the interrupt has started
+	OSIntEnter();
+	CPU_CRITICAL_EXIT();
+	os->pend();
```

Comment:
I guess we went over this in the meeting but I'm still not sure what is supposed to happen in this interrupt. All the stuff you have already done is so the interrupt can run in an RTOS. Also, you should probably post whatever you are pending.

Reply:
Yeah this is a typo I will fix this soon